### DISCUSSION TITLE
Compare server fingerprint - but how?

### DISCUSSION THREAD

**Discussion by ghost** [input-leap/input-leap#2112] [Q&A]
> This issue has been migrated from old Barrier Github repository https://github.com/debauchee/barrier/issues/897
> 
> Issue created on: 2020-10-03 by @ccoenen
> Issue last updated on: 2020-11-14

When I connect a new client, it asks me to verify the server fingerprint. I would love to do so, but I do not see any way to actually accomplish that. The `SSL Fingerprint` field in the gui is empty despite SSL being enabled.

![empty SSL fingerprint label](https://user-images.githubusercontent.com/124909/94987041-b7e3e880-0563-11eb-9093-e9246558a7cd.png)
![SSL is turned on](https://user-images.githubusercontent.com/124909/94987115-10b38100-0564-11eb-97ae-f93f0004fd08.png)

**To Reproduce**

1. Install Barrier 2.3.3 from github's releases page
2. Open
3. see screenshot as above

**Expected behavior**
Have a fingerprint visible.

**Desktop (please complete the following information):**

 - OS: Windows 10
- Barrier version: 2.3.3

**Additional context**

On this machine, I had Synergy installed previously, maybe that interferes somehow?

---

> Commented on: 2020-10-04 by @Ashark

I was going to file exactly this bug.
As I remember, there was not a possibility to see server's fingerprint even at times when this fork was not existent, I mean when I used synergy.

---

> Commented on: 2020-10-04 by @Ashark

I have found a place where it can be checked: $HOME/.local/share/barrier/SSL/Local.txt
I think that should go to readme. And even better, be able to check it from application gui.

Edit: Also, it may be in another location. So to see the current fingerprint, you may run
`cat ~/.local/share/barrier/SSL/Fingerprints/Local.txt`

---

> Commented on: 2020-11-14 by @Ashark

> And even better, be able to check it from application gui.

I have completely missed the fact, that GUI already has the field for that! As seen at the first screenshot.

**Comment by Ayx03** [comment]
I clicked in `Change Settings F4` found that Enable SSL is on by default and it seems like there's nothing I can do so I clicked OK to close the settings, then the both SSL fingerprints appeared.

![image](https://github.com/user-attachments/assets/a261db95-5e51-4852-a2d5-bbca598d71db)
